src/updater/updater.cpp

```diff
@@ -149,446 +95,568 @@ void CheckWeightsConsistency(
+        region.advise(mapped_region::advice_sequential);
+        return region;
+    }
+    catch (const std::exception &e)
+    {
+        util::Log(logERROR) << "Error while trying to mmap " + filename + ": " + e.what();
+        throw;
+    }
 }
 
-EdgeID
-Updater::LoadAndUpdateEdgeExpandedGraph(std::vector<extractor::EdgeBasedEdge> &edge_based_edge_list,
-                                        std::vector<EdgeWeight> &node_weights) const
+tbb::concurrent_vector<GeometryID>
+updateSegmentData(const UpdaterConfig &config,
+                  const extractor::ProfileProperties &profile_properties,
+                  const SegmentLookupTable &segment_speed_lookup,
+                  extractor::SegmentDataContainer &segment_data)
 {
-    // Propagate profile properties to contractor configuration structure
-    extractor::ProfileProperties profile_properties;
-    storage::io::FileReader profile_properties_file(config.profile_properties_path,
-                                                    storage::io::FileReader::HasNoFingerprint);
-    profile_properties_file.ReadInto<extractor::ProfileProperties>(&profile_properties, 1);
     auto weight_multiplier = profile_properties.GetWeightMultiplier();
 
-    if (config.segment_speed_lookup_paths.size() + config.turn_penalty_lookup_paths.size() > 255)
-        throw util::exception("Limit of 255 segment speed and turn penalty files each reached" +
-                              SOURCE_REF);
-
-    util::Log() << "Opening " << config.edge_based_graph_path;
+    std::vector<extractor::QueryNode> internal_to_external_node_map;
+    storage::io::FileReader nodes_file(config.node_based_graph_path,
+                                       storage::io::FileReader::HasNoFingerprint);
+    nodes_file.DeserializeVector(internal_to_external_node_map);
+
+    // vector to count used speeds for logging
+    // size offset by one since index 0 is used for speeds not from external file
+    using counters_type = std::vector<std::size_t>;
+    std::size_t num_counters = config.segment_speed_lookup_paths.size() + 1;
+    tbb::enumerable_thread_specific<counters_type> segment_speeds_counters(
+        counters_type(num_counters, 0));
+    const constexpr auto LUA_SOURCE = 0;
+
+    // The check here is enabled by the `--edge-weight-updates-over-factor` flag it logs a
+    // warning if the new duration exceeds a heuristic of what a reasonable duration update is
+    std::unique_ptr<extractor::SegmentDataContainer> segment_data_backup;
+    if (config.log_edge_updates_factor > 0)
+    {
+        // copy the old data so we can compare later
+        segment_data_backup = std::make_unique<extractor::SegmentDataContainer>(segment_data);
+    }
 
-    auto mmap_file = [](const std::string &filename, boost::interprocess::mode_t mode) {
-        using boost::interprocess::file_mapping;
-        using boost::interprocess::mapped_region;
+    tbb::concurrent_vector<GeometryID> updated_segments;
```

Comment:
Interesting, I will try benchmarking this. Main problem I had so far is that the whole update code has so much variance (+- 200ms) it makes micro-benchmarks hard. :-/